### ISSUE TITLE
BUG: Use object shorthand or index signatures being applied to JavaScript files

### ISSUE THREAD

**Issue by franklygeorgy** [gajus/eslint-plugin-jsdoc#1204]
```warning  Use object shorthand or index signatures instead of `Object`, e.g., `{[key: string]: string}`  jsdoc/check-types``` is being applied to our JavaScript files. Either the example given is wrong for JavaScript or this doesn't apply to JavaScript at all.

We don't want to follow this directive anyways so having an option to toggle it off would be appreciated.

## ESLint Config

```json5
{
    ....
    "rules": {
        "jsdoc/check-types": 1,
    }
}
```

## ESLint sample

```js
/**
 * brand
 *
 * @type {Object.<string, string>}
 */
```

## Environment

- Node version:  v20.10.0
- ESLint version  v8.56.0
- `eslint-plugin-jsdoc` version: 46.0.0

**Comment by brettz9** [comment]
> `` warning Use object shorthand or index signatures instead of `Object`, e.g., `{[key: string]: string}` jsdoc/check-types `` is being applied to our JavaScript files. Either the example given is wrong for JavaScript or this doesn't apply to JavaScript at all.

The example is correct for JavaScript; it denotes an object with string keys which map to string values. While this pattern is not part of the original JSDoc spec, the default mode we use and recommend is the TypeScript flavor of JSDoc. TypeScript tooling *can* be used with plain JavaScript as long as it follows the TypeScript flavor of JSDoc.
 
> We don't want to follow this directive anyways so having an option to toggle it off would be appreciated.

You can disable the TypeScript flavor of JSDoc to use the original flavor by adding the following to your config:

```js
      settings: {
        jsdoc: {
          mode: 'jsdoc',
        },
      }
```

Closing as that should resolve, but feel free to comment further as needed.

**Comment by franklygeorgy** [comment]
Setting mode to `jsdoc` gives us new errors like `Syntax error in type: [Partial<App.Entity.SlaveState>, App.Medicine.Surgery.SimpleReaction]` and `Syntax error in type: import("./setup.js").Settings` because we do use typescript notation in our jsdoc documentation. Again a toggle would be appreciated, but failing that I guess we will just have to add a lot of ignore comments to our documentation.

Take for example `* @type {Object.<string, string>}`. Changing it to `* @type {[Object: string]: string}` creates the error messages `Syntax error in type: [Object: string]: [jsdoc/valid-types]` and `'}' expected` on that line and the error message `Type '{}' is not assignable to type '[Object: string]'` a few lines down at `this.brand = {};`

**Comment by franklygeorgy** [comment]
Never mind, I can't even use ignore comments. Because even though this error message reports that it is cause by `jsdoc/check-types` using `// eslint-disable-next-line jsdoc/check-types` or `// eslint-disable-line jsdoc/check-types` does not make the error go away.

**Comment by brettz9** [comment]
Firstly, you need curly brackets around the object, i.e.,

```js
/**
 * @type {{[Object: string]: string}}
 */
```

And I presume you would want to rename "Object" to "key" or something like that since that portion is indicating the key type. So:

```js
/**
 * @type {{[key: string]: string}}
 */
```

If you actually do need to disable (though I don't think you should), it should be like this:

```js
/* eslint-disable jsdoc/valid-types */
/**
 *
 */
```

**Comment by franklygeorgy** [comment]
Okay, I and my team were misunderstanding how that worked. Adding the second pair of brackets fixes all of our issues.

Thanks for the help!
